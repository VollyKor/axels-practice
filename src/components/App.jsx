import Container from "react-bootstrap/Container";
import { Switch, Route } from "react-router-dom";

import { Cart, Navbar } from "components";

import GlobalStyles from "../styled/globalStyles";

function App() {
  return (
    <>
      <GlobalStyles />
      <Navbar />
      <Switch>
        <Route path="/" exact>
          <Container className="p-3"></Container>
        </Route>
        <Route path="/cart">
          <Cart />
        </Route>
      </Switch>
    </>
  );
}

export default App;
