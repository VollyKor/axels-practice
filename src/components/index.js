export { default as App } from "./App";
export { default as CartLink } from "./CartLink";
export { default as Cart } from "./Cart";
export { default as Navbar } from "./Navbar";
export { default as OrderNav } from "./OrderNav";
export { default as OrderSummary } from "./OrderSummary";
export { default as PriceCounter } from "./PriceCounter";
export { default as ShippingForm } from "./ShippingForm";
