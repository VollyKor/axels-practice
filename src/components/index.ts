export { default as App } from './App';
export { default as Cart } from './Cart';
export { default as CartLink } from './CartLink';
export { default as CityInput } from './CityInput';
export { default as Conditions } from './Conditions';
export { default as CreditCardIcon } from './CreditCardIcon';
export { default as GoodleModal } from './GoodleModal';
export { default as Navbar } from './Navbar';
export { default as NavBarLink } from './NavBarLink';
export { default as OrderData } from './OrderData';
export { default as OrderForms } from './OrderForms';
export { default as OrderNav } from './OrderNav';
export { default as OrderSummary } from './OrderSummary';
export { default as PriceCounter } from './PriceCounter';
export { default as PrintPage } from './PrintPage';
export { default as ProductImg } from './ProductImg';
export { default as ProductInfo } from './ProductInfo';
export { default as Tooltip } from './Tooltip';

// Forms
export { default as ShippingForm } from './forms/ShippingForm';
export { default as BillingForm } from './forms/BillingForm';
export { default as PaymentForm } from './forms/PaymentForm';
