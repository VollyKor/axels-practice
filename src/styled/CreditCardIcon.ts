import styled from 'styled-components';

export const IconWrapper = styled.div`
    position: absolute;
    top: 8px;
    right: 5px;
`;

export const Icon = styled.img`
    display: block;

    height: 24px;
`;
