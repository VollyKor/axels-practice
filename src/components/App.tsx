import Container from 'react-bootstrap/Container';
import { Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { Cart, Navbar } from 'components';
import PrintPage from './PrintPage';

import GlobalStyles from '../styled/utils/globalStyles';
import theme from '../styled/utils/themeStyles';
import TestCounter from './practice/TestCounter';
import { AnimateItem } from './practice/TestTransition';

const App = (): JSX.Element => (
    <ThemeProvider theme={theme}>
        <GlobalStyles />

        <Switch>
            <Route path="/print" exact>
                <PrintPage />
            </Route>
            <Route path="/">
                <Navbar />
                <Route path="/" exact>
                    <Container className="p-3">
                        <AnimateItem />
                        <TestCounter />
                    </Container>
                </Route>
                <Route path="/cart">
                    <Cart />
                </Route>
            </Route>
        </Switch>
    </ThemeProvider>
);

export default App;
