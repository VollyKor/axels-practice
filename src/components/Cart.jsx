import { Row, Col, Container } from "react-bootstrap";

import { ShippingForm, OrderNav, OrderSummary } from "components";

export default function Cart() {
  return (
    <Container>
      <Col lg={9} className="mx-auto ">
        <Row className="mx-auto bg-white shadow rounded">
          <Col lg={7} className="px-4">
            <OrderNav />
            <ShippingForm />
          </Col>
          <Col lg={5} className="position-relative bg-light rounded">
            <OrderSummary />
          </Col>
        </Row>
      </Col>
    </Container>
  );
}
