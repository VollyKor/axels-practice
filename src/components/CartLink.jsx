import { CartWrapper, CartText, CartNotification } from '../styled/CartLink';
import { IoCartOutline } from 'react-icons/io5';

const CartLink = ({ productsInCart = 0 }) => (
    <CartWrapper>
        <CartText className="mr-1">cart</CartText>
        <IoCartOutline size="25px" />
        <CartNotification className=" d-flex justify-content-center align-items-center">
            <span>{productsInCart}</span>
        </CartNotification>
    </CartWrapper>
);

export default CartLink;
