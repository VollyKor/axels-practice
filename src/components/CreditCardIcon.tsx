import visaSrc from 'images/visa-icon.png';
import masterCardSrc from 'images/mastercard-icon.png';
import discoverCardSrc from 'images/discover-icon.png';

import { Icon, IconWrapper } from 'styled/CreditCardIcon';

interface Props {
    value: string;
}

const CreditCardIcon = ({ value }: Props) => {
    return (
        <IconWrapper>
            {value[0] === '4' && <Icon src={visaSrc} alt="visa icon" />}

            {value[0] === '5' && (
                <Icon src={masterCardSrc} alt="mastercard icon" />
            )}

            {value[0] === '6' && (
                <Icon src={discoverCardSrc} alt="mastercard icon" />
            )}
        </IconWrapper>
    );
};

export default CreditCardIcon;
