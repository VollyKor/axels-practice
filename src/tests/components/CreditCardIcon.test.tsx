import { shallow } from 'enzyme';

import { CreditCardIcon } from 'components';
import { Icon } from 'styled/CreditCardIcon';

import visaSrc from 'images/visa-icon.png';
import masterCardSrc from 'images/mastercard-icon.png';
import discoverCardSrc from 'images/discover-icon.png';

it('renders snapshot', () => {
    const wrapper = shallow(<CreditCardIcon value="5" />);

    expect(wrapper).toMatchSnapshot();
});

it('render correct VisaIcon', () => {
    const wrapper = shallow(<CreditCardIcon value="4" />);

    const img = wrapper.find(Icon);

    expect(img.prop('src')).toEqual(visaSrc);
});

it('render correct MasterCardIcon', () => {
    const wrapper = shallow(<CreditCardIcon value="5" />);

    const img = wrapper.find(Icon);

    expect(img.prop('src')).toEqual(masterCardSrc);
});

it('render correct DiscoverCardIcon', () => {
    const wrapper = shallow(<CreditCardIcon value="6" />);

    const img = wrapper.find(Icon);

    expect(img.prop('src')).toEqual(discoverCardSrc);
});
