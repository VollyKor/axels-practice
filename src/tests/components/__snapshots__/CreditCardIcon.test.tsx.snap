// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<styled.div>
  <styled.img
    alt="mastercard icon"
    src="mastercard-icon.png"
  />
</styled.div>
`;
